build/Build.proj

```diff
@@ -0,0 +1,44 @@
+<Project DefaultTargets="Build;Test;Pack">
+
+  <!-- Because this project does not reference any SDK,
+       the following .props file is not auto-imported: -->
+  <Import Project="..\Directory.Build.Props" />
+
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Release</Configuration>
+    <CommonBuildProperties>WarningLevel=0;NoWarn=1591;RunCodeAnalysis=false;Configuration=$(Configuration)</CommonBuildProperties>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Solution Include="$(RootDirectory)Moq.sln"/>
+    <Source Include="$(RootDirectory)src\Moq\Moq.csproj" />
+    <TestProjects Include="$(RootDirectory)tests\**\*.*proj" />
+  </ItemGroup>
+    
+  <Target Name="Clean">
+    <MSBuild Projects="@(Solution)" Properties="$(CommonBuildProperties)" Targets="Clean" />
+    <Exec Command="rmdir $(OutputDirectory) /S /Q" Condition="Exists('$(OutputDirectory)')" ContinueOnError="true" />
+  </Target>
+
+  <Target Name="Rebuild" DependsOnTargets="Clean;Build" />
+    
+  <Target Name="Build" DependsOnTargets="Configure">
+    <MSBuild Projects="@(Solution)" Properties="$(CommonBuildProperties)" />
+  </Target>
+
+  <Target Name="Test" DependsOnTargets="Build">
+    <MSBuild Projects="@(TestProjects)" Targets="Test" Properties="$(CommonBuildProperties)" BuildInParallel="False" />
+  </Target>
+
+  <Target Name="Pack" DependsOnTargets="Build">
+    <MSBuild Projects="@(Source)" Targets="Pack" Properties="$(CommonBuildProperties)" BuildInParallel="False" />
+  </Target>
+
+  <Target Name="Configure">
+    <MSBuild Projects="$(RootDirectory)Moq.sln" Targets="Restore" Properties="Step=1;$(CommonBuildProperties)" />
```

Comment:
Not sure what those `Step` properties passed down are for.

Reply:
They are a hack to prevent MSBuild from caching information about the invoked projects. The thing is that a `Restore` will create a `.props` file under `obj/` which includes build files from the NuGet packages (e.g. GitInfo). If you invoke `Restore` together with another task, MSBuild won't read the auto-generated `.props` file and won't find the NuGet package-provided targets.

I'd love to get rid of this.